main.tf

```diff
@@ -115,17 +116,20 @@ resource "aws_codepipeline" "pipeline" {
     }
   }
 
-  stage {
-    name = "Deploy"
-    action {
-      category        = "Deploy"
-      name            = "Deploy"
-      owner           = "AWS"
-      provider        = var.deploy_provider
-      version         = "1"
-      input_artifacts = ["build_output"]
-
-      configuration = var.deploy_configuration
+  dynamic "stage" {
+    for_each = local.has_deploy_stage ? [1] : []
```

Comment:
You aren't actually iterating over anything here, so `count` would probably be a little more understandable:
```
count = local.has_deploy_stage ? 1 : 0
```

Reply:
I believe the dynamic block requires a `for_each` and not a `count`